lib/modules/datasource/unity3d/index.ts

```diff
@@ -0,0 +1,73 @@
+import { XmlDocument } from 'xmldoc';
+import { cache } from '../../../util/cache/package/decorator';
+import * as Unity3dVersioning from '../../versioning/unity3d';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+export class Unity3dDatasource extends Datasource {
+  static readonly streams: { [key: string]: string } = {
+    lts: 'https://unity.com/releases/editor/lts-releases.xml',
+    stable: 'https://unity.com/releases/editor/releases.xml',
+    beta: 'https://unity.com/releases/editor/beta/latest.xml',
+  };
+
+  static readonly id = 'unity3d';
+
+  override readonly defaultRegistryUrls = [
+    Unity3dDatasource.streams.stable,
+    Unity3dDatasource.streams.lts,
+  ];
+
+  override readonly defaultVersioning = Unity3dVersioning.id;
+
+  override readonly registryStrategy = 'merge';
+
+  constructor() {
+    super(Unity3dDatasource.id);
+  }
+
+  getByStream = async (
+    registryUrl: string | undefined,
+    withHash: boolean,
+  ): Promise<ReleaseResult | null> => {
+    const response = await this.http.get(registryUrl!);
+    const document = new XmlDocument(response.body);
```

Comment:
These can throw exceptions

Reply:
Wrapped them and added tests inside a439771351614f05a9b6ad7d2d525941e83e11bc. I noticed `handleGenericErrors()` exists, but wasn't sure if it's of any use here.